source/common/upstream/maglev_lb.cc

```diff
@@ -3,17 +3,22 @@
 namespace Envoy {
 namespace Upstream {
 
-MaglevTable::MaglevTable(const HostVector& hosts) {
+MaglevTable::MaglevTable(const HostVector& hosts, uint64_t table_size) : table_size_(table_size) {
+  ASSERT(Primes::isPrime(table_size));
```

Comment:
would it be better to throw an EnvoyException here instead of the assert?

Reply:
The error checking is a little tricky in terms of who would catch the exception. Right now it's only configurable in tests. I will add a comment about adding error checking in the future if we allow the table size to be configured.